xcdat/dataset.py

```diff
@@ -387,7 +399,10 @@ def _preprocess(
         ds_new = callable(ds)
 
     if decode_times:
-        ds_new = decode_time(ds_new)
+        try:
+            ds_new = decode_time(ds_new)
+        except KeyError as e:
+            logger.warning(e)
```

Comment:
We can change the level to `.debug` if we want to avoid printing these warnings to the console. This is reasonable because the default behavior is `decode_times=True` and if we expect `decode_times=True` to work in _most_ cases.

One scenario to consider is if the dataset has time coordinates without CF attributes (e.g., `axis` or `standard_name`) and the user expects the coordinates to be decodable. In this case, the message won't be raised unless the user sets the xcdat logger level to `.debug`, or if they try to run temporal averaging operations which will throw the `KeyError`.

Let me know what you think about the above scenario and how frequent you think that might occur.

Reply:
I think you've convinced me we should keep the code as you have it. @lee1043 - thoughts?